src/puppetlabs/trapperkeeper/services/status/status_service.clj

```diff
@@ -33,6 +33,11 @@
       (add-ring-handler this handler))
     context)
 
+  (stop [this context]
+    (status-core/dissoc-status-context! (:status-fns context)
+                                        status-core/status-service-name)
+    context)
```

Comment:
Couple of things here:

1. Is there any reason not to just clear the whole atom here?
2. The restart logic in TK involves throwing away the service's context entirely and replacing it with a clean one when it starts the service back up.  So I'm curious what kind of code you were running where that didn't sufficiently clear things out?